src/VirtualClient/VirtualClient.Dependencies/AMDGPUDriverInstallation.cs

```diff
@@ -107,17 +141,192 @@ await this.stateManager.SaveStateAsync(nameof(AMDGPUDriverInstallation), new Sta
+                                // different distro installation to be addded.
+                                this.Logger.LogMessage(
+                                    $"AMD GPU driver installation is not supported by Virtual Client on the current Linux distro '{linuxDistributionInfo.LinuxDistribution}'.", 
+                                    telemetryContext);
+
+                                break;
+                        }
+
+                        await this.InstallAMDGPUDriverLinux(linuxDistributionInfo.LinuxDistribution, telemetryContext, cancellationToken)
+                                   .ConfigureAwait(false);
+
+                        await this.stateManager.SaveStateAsync(nameof(AMDGPUDriverInstallation), new State(), cancellationToken)
+                            .ConfigureAwait(false);
                     }
 
                     VirtualClientRuntime.IsRebootRequested = this.RebootRequired;
                 }
+
+                if (this.Platform == PlatformID.Unix)
+                {
+                    LinuxDistributionInfo linuxDistributionInfo = await this.systemManager.GetLinuxDistributionAsync(cancellationToken)
+                        .ConfigureAwait(false);
+
+                    await this.ExecutePostRebootCommands(linuxDistributionInfo.LinuxDistribution, telemetryContext, cancellationToken)
+                        .ConfigureAwait(false);
+                }
             });
         }
 
-        private async Task InstallAMDGPUDriver(EventContext telemetryContext, CancellationToken cancellationToken)
+        [SuppressMessage("StyleCop.CSharp.ReadabilityRules", "SA1118:Parameter should not span multiple lines", Justification = "Readability")]
+        private async Task InstallAMDGPUDriverLinux(LinuxDistribution linuxDistribution, EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(this.LinuxInstallationFile))
+            {
+                throw new DependencyException($"The linux installation file can not be null or empty and it is: {this.LinuxInstallationFile}", ErrorReason.DependencyNotFound);
+            }
+
+            // The .bashrc file is used to define commands that should be run whenever the system
+            // is booted. For the purpose of the AMD GPU driver installation, we want to include extra
+            // paths in the $PATH environment variable post installation.
+            string bashRcPath = $"/home/{this.Username}/.bashrc";
+
+            // We hit a bug where the .bashrc file does not exist on the system. To prevent issues later
+            // we are creating the file if it is missing.
+            if (!this.fileSystem.File.Exists(bashRcPath))
+            {
+                await this.fileSystem.File.WriteAllLinesAsync(
+                    bashRcPath,
+                    new string[]
+                    {
+                        "# ~/.bashrc: executed by bash(1) for non-login shells.",

```

Comment:
is this just comments? do we have to write it?

Reply:
actually, whenever on any linux system, bashrc file is created, these lines are present, so to keep that consistency of linux, keeping these lines.